Cannot install gcc: C compiler cannot create executables

When trying to install gcc using spack I get an error when compiling the actual gcc source code. I get the same error for multiple machines (our cluster and my own private laptop) and for at least versions 6.1.0 and 5.4.0.

The error is `C compiler cannot create executables` but I don't know how to fix it as in [issue 1038](https://github.com/LLNL/spack/issues/1038).

The file `spack-build.out` gives:

    ==> '/home/toon/testspack/var/spack/stage/gcc-6.1.0-ch6rh6557ua3p7fnrauyqswcpxqn4bt6/gcc-6.1.0/configure' '--prefix=/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/gcc-6.1.0-ch6rh6557ua3p7fnrauyqswcpxqn4bt6' '--libdir=/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/gcc-6.1.0-ch6rh6557ua3p7fnrauyqswcpxqn4bt6/lib64' '--disable-multilib' '--enable-languages=c,java,c++,fortran,objc,go' '--with-mpc=/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/mpc-1.0.3-u3edu7j3u22injou273z4hizip6vhjjt' '--with-mpfr=/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/mpfr-3.1.4-zktmbe3mgwuw3t4c3l7gb7ulnxgzd2tb' '--with-gmp=/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/gmp-6.1.1-36t7wx6655klu7waooeymnz46rxynaug' '--enable-lto' '--with-quad' '--with-sysroot=/' '--with-stage1-ldflags=-Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/gcc-6.1.0-ch6rh6557ua3p7fnrauyqswcpxqn4bt6/lib -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/gcc-6.1.0-ch6rh6557ua3p7fnrauyqswcpxqn4bt6/lib64 -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/gmp-6.1.1-36t7wx6655klu7waooeymnz46rxynaug/lib -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/isl-0.14-b7nimaueyc7lzqpmdkttmecgm7srhc75/lib -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/mpc-1.0.3-u3edu7j3u22injou273z4hizip6vhjjt/lib -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/binutils-2.25-e34u7ath72ak6d6sicerzub37dvhubf5/lib -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/mpfr-3.1.4-zktmbe3mgwuw3t4c3l7gb7ulnxgzd2tb/lib -static-libstdc++ -static-libgcc' '--with-boot-ldflags=-Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/gcc-6.1.0-ch6rh6557ua3p7fnrauyqswcpxqn4bt6/lib -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/gcc-6.1.0-ch6rh6557ua3p7fnrauyqswcpxqn4bt6/lib64 -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/gmp-6.1.1-36t7wx6655klu7waooeymnz46rxynaug/lib -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/isl-0.14-b7nimaueyc7lzqpmdkttmecgm7srhc75/lib -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/mpc-1.0.3-u3edu7j3u22injou273z4hizip6vhjjt/lib -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/binutils-2.25-e34u7ath72ak6d6sicerzub37dvhubf5/lib -Wl,-rpath,/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/mpfr-3.1.4-zktmbe3mgwuw3t4c3l7gb7ulnxgzd2tb/lib -static-libstdc++ -static-libgcc' '--with-gnu-ld' '--with-ld=/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/binutils-2.25-e34u7ath72ak6d6sicerzub37dvhubf5/bin/ld' '--with-gnu-as' '--with-as=/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/binutils-2.25-e34u7ath72ak6d6sicerzub37dvhubf5/bin/as' '--with-isl=/home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/isl-0.14-b7nimaueyc7lzqpmdkttmecgm7srhc75'
    checking build system type... x86_64-pc-linux-gnu
    checking host system type... x86_64-pc-linux-gnu
    checking target system type... x86_64-pc-linux-gnu
    checking for a BSD-compatible install... /usr/bin/install -c
    checking whether ln works... yes
    checking whether ln -s works... yes
    checking for a sed that does not truncate output... /bin/sed
    checking for gawk... gawk
    checking for libatomic support... yes
    checking for libcilkrts support... yes
    checking for libitm support... yes
    checking for libsanitizer support... yes
    checking for libvtv support... yes
    checking for libmpx support... yes
    checking for gcc... /home/toon/testspack/lib/spack/env/gcc/gcc
    checking for C compiler default output file name... 
    configure: error: in `/tmp/toon/spack-stage/spack-stage-d1WhBG/spack-build':
    configure: error: C compiler cannot create executables
    See `config.log' for more details.

and [config.log](https://github.com/LLNL/spack/files/410994/config.log.txt) is attached. It looks like the error occurs here:

    configure:4436: /home/toon/testspack/lib/spack/env/gcc/gcc    conftest.c  >&5
    /home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/binutils-2.25-e34u7ath72ak6d6sicerzub37dvhubf5/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/crti.o: unrecognized relocation (0x2a) in section `.init'
    /home/toon/testspack/opt/spack/linux-Ubuntu16-x86_64/gcc-5.4.0/binutils-2.25-e34u7ath72ak6d6sicerzub37dvhubf5/bin/ld: final link failed: Bad value

Do you have any idea how to solve this? Thank in advance.

--- a/var/spack/repos/builtin/packages/gcc/package.py
+++ b/var/spack/repos/builtin/packages/gcc/package.py
@@ -69,7 +69,7 @@ class Gcc(AutotoolsPackage):
             multi=True,
             description='Compilers and runtime libraries to build')
     variant('binutils',
-            default=sys.platform != 'darwin',
+            default=False,
             description='Build via binutils')
     variant('piclibs',
             default=False,

